packages/api/amplify_api/example/integration_test/graphql/iam_test.dart

```diff
@@ -226,7 +226,90 @@ void main({bool useExistingTestUser = false}) {
             await Amplify.API.query(request: reqThatShouldWork).response;
         expect(res, hasNoGraphQLErrors);
         expect(res.data?.name, testName);
-        await deleteBlog(res.data!.id);
+        await deleteBlog(res.data!);
+      });
+
+      testWidgets(
+          'should GET a model with custom primary key and complex identifier using model helpers',
+          (WidgetTester tester) async {
+        const name = 'Integration Test CpkParent to fetch';
+        final cpkParent = await addCpkParent(name);
+        final req = ModelQueries.get(
+          CpkOneToOneBidirectionalParentCD.classType,
+          cpkParent.modelIdentifier,
+        );
+        final res = await Amplify.API.query(request: req).response;
+        final data = res.data;
+        expect(res, hasNoGraphQLErrors);
+        expect(data, equals(cpkParent));
+      });
+
+      testWidgets(
+          'should GET a child and include parent with complex identifier and custom primary key',
+          (WidgetTester tester) async {
```

Comment:
nit: Following up on our offline discussion. This could be a test case for a relationship diagram-like comment.

Reply:
thanks, will work on improving

Reply:
added small comment